File: php/src/Advertisement/Infrastructure/Errors/ZeroRecordsError.php
<?php
declare(strict_types=1);

namespace Demo\App\Advertisement\Infrastructure\Errors;

use Demo\App\Common\Result;

final readonly class ZeroRecordsError
{
    public static function build(string $id): Result
    {
        return Result::failure('Advertisement not found with ID: ' . $id, 'NOT_FOUND');
    }
}

File: kotlin/src/main/kotlin/advertisement/infrastructure/errors/ZeroRecordsError.kt
package advertisement.infrastructure.errors

import common.infrastructure.InfrastructureError

class ZeroRecordsError private constructor(val value: String, errorMessage: String) : InfrastructureError(errorMessage) {
    companion object {
        fun withId(value: String): ZeroRecordsError {
            return ZeroRecordsError(value, "Advertisement not found with Id $value")
        }
    }
}
